File: backend/CarsIntegration/CarsIntegration/Controllers/Car.cs
namespace CarsIntegration.Controllers
{
    internal class Car
    {
        public int CarID { get; set; }
        public string Manufacturer { get; set; }
        public string Model { get; set; }
        public string TransmissionType { get; set; }
        public int NoOfSeats { get; set; }
        public int EngineCapacity { get; set; }
        public string CarType { get; set; }
        public int YearOfManufacture { get; set; }
        public string FuelType { get; set; }
        public int Mileage { get; set; }
        public bool Availability { get; set; }
        public decimal CostPerHour { get; set; }
        public string LocationOfCar { get; set; }

        public decimal TotalCost { get; set; }
    }

    public class Reservation
    {
        public int ReservationID { get; set; }
        public int CarID { get; set; }
        public DateTime ReservationStart { get; set; }
        public DateTime ReservationEnd { get; set; }
        public decimal TotalCarCost { get; set; }
        public string Status { get; set; }
        public double CostPerHour { get; internal set; }
    }
}

File: frontend/src/app/shared/model/car/Car.ts
export interface Car {
    availability: boolean;
    carID: number;
    
    carType: string | null;
    costPerHour: number;
    engineCapacity: number;
    fuelType: string | null;
    locationOfCar: string | null;
    manufacturer: string;
    mileage: number;
    model: string;
    noOfSeats: number;
    transmissionType: string;
    yearOfManufacture: number;
    totalCost: number;
    averageRating?: number; 
    imageUrl?: string;
  }


  export interface Reservation {
    carID: number;
    reservationStart: Date; // ISO formatted date string
    reservationEnd: Date; // ISO formatted date string
    totalCarCost: number;
    costPerHour: number;
    status: string; // 'booked' or 'pending'
    userID:number;
  }
  export interface Reservation2 {
    carID: number;
    reservationID:number
    reservationStart: Date; // ISO formatted date string
    reservationEnd: Date; // ISO formatted date string
    totalCarCost: number;
    costPerHour: number;
    status: string; // 'booked' or 'pending'
    userID: number;
    carDetails: Car; // Adding the carDetails property
   }

   export interface Bill {
    reservationID: number;
    userID: number;
    carID: number;
    reservationStart: Date;
    reservationEnd: Date;
    totalCarCost: number;
  }

  export interface Bill2 {
    billID: number;
    reservationID: number;
    userID: number;
    carID: number;
    reservationStart: Date;
    reservationEnd: Date;
    totalCarCost: number;
    carDetails: Car;
  }

  export interface Review {
    RatingID?: number;
    rating: number;
    reviewText: string;
    userID: number;
    carID: number;
  }

  export class Image {
    imageID: number;
    imageUrl: string;
    carID: number;
  }